### ISSUE TITLE
Data center compatibility

### ISSUE THREAD

**Issue by johnrc** [monitorjbl/stash-token-auth#9]
Any idea what it'd take to make this plugin [cluster or data center compatible](https://developer.atlassian.com/stash/docs/latest/how-tos/cluster-safe-plugins.html#Marking_your_plugin_as_cluster_compatible_for_the_Marketplace)? Looking at the code, settings are persisted using `PluginSettings`. That seems to save to the database which would be safe and usable for a cluster.

**Comment by monitorjbl** [comment]
I suspect the plugin will work on Datacenter distributions just fine. The problem with making it officially compatible is that I don't have a way of testing it and making sure that it always stays compatible.

**Comment by johnrc** [comment]
Makes sense. I'll give it a shot and see how it goes. The [Confluence documentation](https://developer.atlassian.com/confdev/development-resources/confluence-developer-faq/how-do-i-ensure-my-add-on-works-properly-in-a-cluster/starting-a-confluence-cluster-on-a-single-machine) makes it sound easy to test plugins on a cluster. I suspect it's similar for Stash but I don't know enough about it.

**Comment by ebukoski** [comment]
Has anyone tried to modify this plugin for Data Center compatibility?

**Comment by monitorjbl** [comment]
To my knowledge no, but it may work in that environment. I don't have one to test with, sadly.

**Comment by ebukoski** [comment]
The plugin did install on Data Center but with a warning and "INCOMPATIBLE" badge.

I'm planning on testing and will report back here if I find any issues.

**Comment by kutsal** [comment]
I just created a PR (#21) for this which fixes a potential issue fetching plugin configuration given a Data Center setup.

**Comment by ebukoski** [comment]
@monitorjbl Can we get PR #21 merged to support the Data Center users?
